runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollOverlay.java

```diff
@@ -22,74 +23,321 @@
+
+			int width = 150;
+
+			panelComponent.setTitle("Cryptic Clue");
+			panelComponent.setWidth(width);
+
+			panelComponent.getLines().add(new PanelComponent.Line("Clue:", ""));
+
+			for (String line : lineBreakText(crypticClue.getText(),
+					width - PanelComponent.LEFT_BORDER - PanelComponent.RIGHT_BORDER,
+					graphics.getFontMetrics()))
+			{
+				panelComponent.getLines().add(new PanelComponent.Line(line, TITLED_CONTENT_COLOR, "", TITLED_CONTENT_COLOR));
+			}
+
+			if (crypticClue.getNpc() != null)
+			{
+				panelComponent.getLines().add(new PanelComponent.Line("NPC:", ""));
+				panelComponent.getLines().add(new PanelComponent.Line(crypticClue.getNpc(), TITLED_CONTENT_COLOR, "", Color.WHITE));
+			}
+
+			if (crypticClue.hasObject())
+			{
+				ObjectComposition object = client.getObjectDefinition(crypticClue.getObjectId());
+
+				if (object != null)
+				{
+					panelComponent.getLines().add(new PanelComponent.Line("Object:", ""));
+					panelComponent.getLines().add(new PanelComponent.Line(object.getName(), TITLED_CONTENT_COLOR, "", Color.WHITE));
+				}
+			}
+
+			panelComponent.getLines().add(new PanelComponent.Line("Solution:", ""));
+
+			for (String line : lineBreakText(crypticClue.getSolution(),
+					width - PanelComponent.LEFT_BORDER - PanelComponent.RIGHT_BORDER,
+					graphics.getFontMetrics()))
+			{
+				panelComponent.getLines().add(new PanelComponent.Line(line, TITLED_CONTENT_COLOR, "", Color.WHITE));
+			}
 		}
-		else
+		else if (clue instanceof EmoteClue)
 		{
-			for (int id : clue.getIds())
+			EmoteClue emoteClue = (EmoteClue) clue;
+
+			panelComponent.setTitle("Emote Clue");
+
+			panelComponent.getLines().add(new PanelComponent.Line("Emotes:", ""));
+			panelComponent.getLines().add(new PanelComponent.Line(emoteClue.getFirstEmote().getName(), TITLED_CONTENT_COLOR, "", TITLED_CONTENT_COLOR));
+			if (emoteClue.getSecondEmote() != null)
 			{
-				ItemComposition clueItem = client.getItemDefinition(id);
+				panelComponent.getLines().add(new PanelComponent.Line(emoteClue.getSecondEmote().getName(), TITLED_CONTENT_COLOR, "", TITLED_CONTENT_COLOR));
+			}
+
+			if (emoteClue.getItemIds().length != 0)
+			{
+				panelComponent.setWidth(160);
+
+				panelComponent.getLines().add(new PanelComponent.Line("Equip:", ""));
 
-				if (clueItem == null)
+				Set<Integer> equippedItems = getEquippedItems();
+
+				for (int itemId : emoteClue.getItemIds())
 				{
-					continue;
+					ItemComposition itemDefinition = client.getItemDefinition(itemId);
+
+					if (itemDefinition != null)
+					{
+						if (equippedItems.contains(itemId))
+						{
+							panelComponent.getLines().add(new PanelComponent.Line(itemDefinition.getName(), TITLED_CONTENT_COLOR, "X", Color.GREEN));
+						}
+						else
+						{
+							panelComponent.getLines().add(new PanelComponent.Line(itemDefinition.getName(), TITLED_CONTENT_COLOR, "-", Color.RED));
+						}
+					}
 				}
+			}
+			else
+			{
+				panelComponent.setWidth(130);
```

Comment:
What is the reasoning for setting different width for each clue type? Can't they share common max width for consistency?

Reply:
Not really, as some of the clue types that use both the left and right side don't produce a lot of text, which makes it look silly when the box is extremely large in relation to the width of the text.